theano/gof/opt.py

```diff
@@ -530,10 +528,10 @@ def process_node(self, fgraph, node):
             if len(node.inputs) != len(candidate.inputs):
                 continue
 
-            inputs_match = all(node_in is cand_in
-                               for node_in, cand_in in zip(node.inputs,
-                                                           candidate.inputs))
-            if inputs_match and node.op == candidate.op:
+            if node.op != candidate.op:
+                continue
```

Comment:
Could it be possible that `not node.op == candidate.op` and `node.op != candidate.op` give different results (for scan in particular)?